Fix mach0 endianness

wargio opened the pull request:
# DO NOT SQUASH

**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Tries to fix mach0 endianness

wargio committed:
Partially refactor mach0 to avoid rz_buf_fread() and others

thestr4ng3r commented:
For endian `rz_buf_fread_at` is usually not the issue since it is aware of the endian. But I still like seeing it removed since it is very easily misused and evades type-checking.

wargio commented:
yeah, it doesn't fix the endianness :/
i think the issue is on maps, but i don't know where.

thestr4ng3r commented:
No worries, I think I already found it, will check in a moment.

thestr4ng3r committed:
Fix endian in macho reloc parsing

thestr4ng3r commented:
That seems to fix half of them, but please don't merge yet since I want to test it on a ppc macho.

thestr4ng3r committed:
Fix some db/formats/mach0 tests on hosts with asm.bits=32

Fatmacho loading selects the file based on both arch and bits, so if the
preselected asm.bits was not 64 (which is the case on OpenBSD/sparc64
for some reason), these tests would fail.

thestr4ng3r commented:
This does it at least on sparc64. I have seen two more failures in `db/formats/mach0/fatmach0` on ppc Mac OS X, but that can be addressed separately. Not sure if they happen on system-z too.

XVilka commented on librz/bin/format/mach0/mach0_relocs.c:
```diff
@@ -10,6 +10,37 @@
+static void read_relocation_info(struct relocation_info *dst, ut8 *src, bool big_endian) {
+	dst->r_address = (st32)rz_read_at_ble32(src, 0, big_endian);
+	uint32_t field = rz_read_at_ble32(src, 4, big_endian);
+	if (big_endian) {
```
QEMU has helper macroses like `EXTRACT` etc. We could think about something similar too.

XVilka reviewed (approved):
Looks good. Lets see what Travis says.

XVilka commented:
17 XX now.
